apps/shared/models/base.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Because it is less expensive to store strings than Firestore references, with no change in the
+ * value it provides, we have this local type to express this usage.
+ */
+export type FirestoreReference<T> = string;
```

Comment:
Nit: Should we use a branded type so that we can make this type-safe. Can be a follow-up

```suggestion
export type FirestoreReference<T> = {
     // ensures this "Reference" string is not passed around as "string" accidentally.
  __eefirestoreReference: true;
};

export function getValueOfReference<T>(ref: FirestoreReference<T>): T {
  // we know this is a string.
  const value = ref as string;
  
  // do something
}
```

Reply:
I played with this a bit and I think we should do it in a follow up as it makes the question of how we do some of the nested interactions a bit tricky and I think when we are in a place of more practical decisions being made it will be clearer.